plugins/executor/slack/plugin.yaml

```diff
@@ -0,0 +1,61 @@
+kind: ExecutorPlugin
+apiVersion: argoproj.io/v1alpha1
+metadata:
+  name: slack
+  annotations:
+    workflows.argoproj.io/description: |
+      This plugin sends a Slack message.
+
+      You must create a secret:
+
+      ```yaml
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: slack-executor-plugin
+      stringData:
+        URL: https://hooks.slack.com/services/.../.../...
+      ```
+
+      Example:
+
+      ```yaml
+      apiVersion: argoproj.io/v1alpha1
+      kind: Workflow
+      metadata:
+        generateName: slack-example-
+      spec:
+        entrypoint: main
+        templates:
+          - name: main
+            plugin:
+              slack:
+                text: "{{workflow.name}} finished!"
```

Comment:
We should require users to provide their schema when building plugins for validation, etc. Nowhere other than this example is this `text` field documented.

Reply:
v2?